code/unclassified/src/apriori/apriori.py

```diff
@@ -0,0 +1,131 @@
+# Part of Cosmos by OpenGenus Foundation
+# Find frequently occuring subsets of items in a set of transactions
+# Contributed by: Aditya Lohana (dundermiflin)
+
+
+import csv
+from itertools import combinations, chain
+
+
+min_support = 0.1
+
+
+def csv_to_list(filename):
+	"""
+		Function to convert the transaction-dataset into a list of lists
+	"""
+	with open(filename, 'r') as f:
+		reader = csv.reader(f)
+		data = [row for row in reader]
+		return data
+
+
+def count_support(transactions):
+	"""
+		Function to count the support of each unique item in the dataset
+		Returns a dictionary with item names as keys, and support as values
+	"""
+	sup = {}
+	num_transactions = len(transactions)
+	for row in transactions:
+		for col in row:
+			if col not in sup.keys():
+				sup[col] = 1
+			else:
+				sup[col] += 1
+	for item in sup.keys():
+		sup[item] /= num_transactions
+	return sup
+
+
+def join_itemset(a, b):
+	"""
+		Function to merge to length-N itemsets with (N-1)-length common prefix
+		Returns an itemset of length N+1
+	"""
+	a = list(a)
+	b = list(b)
+	x = a[: -1]
+	y = b[: -1]
+	if x == y:
+		x.extend([a[-1], b[-1]])
+		return tuple(x)
+	return None
+
+
+def subsets(itemlist):
+	"""
+		Function to generate all subsets of a given itemset
+	"""
+	subs = []
+	for i in range(1, len(itemlist)):
+		subs.extend(list(set(combinations(itemlist, i))))
+	return subs
+
+
+def frequent_candidates(transactions, candidates):
+	"""
+		Function to count support of candidate itemsets,
+		and filter out ones which do not meet the criteria.
+		Returns a list of frequent itemsets from the candiates,
+		and a dictionary containing their support.
+	"""
+	count = {}
+	freq = {}
+	for c in candidates:
+		count[c] = 0
+	for t in transactions:
+		for c in candidates:
+			if set(c) <= set(t):
+				count[c] += 1
+	for c in count:
+		if count[c] >= min_support*len(transactions):
+			freq[c] = count[c]/len(transactions)
+	return freq.keys(), freq
```

Comment:
PEP8 says four space indent; Please follow this rule.

Reply:
I set the default indent to 4 spaces in my iDE. Let me know if there's some issue. I'll make changes accordingly.